whitepapers/Tables.md

```diff
@@ -1,5 +1,114 @@
+
+## Ingest Configuration
+
+### IngestConfig
+Describes one way to harvest metadata from a Source, and how to transform the result.
+
+| Column | Type | Indexed | Nullable | FK | Default | Description |
+|:-------|:----:|:-------:|:--------:|:--:|:-------:|:------------|
+| `source_id` | int |  |  | ✓ | | Source to harvest from |
+| `base_url` | text |  |  |  | | URL of the API or endpoint where the metadata is available |
+| `earliest_date` | date |  | ✓ |  | | Earliest date with available data |
+| `rate_limit_allowance` | int |  |  |  | 5 | Number of requests allowed every `rate_limit_period` seconds |
+| `rate_limit_period` | int |  |  |  | 1 | Number of seconds for every `rate_limit_allowance` requests |
+| `harvester_id` | int |  |  | ✓ | | Harvester to use |
+| `harvester_kwargs` | jsonb |  | ✓ |  | | JSON object passed to the harvester as kwargs |
+| `transformer_id` | int |  |  | ✓ | | Transformer to use |
+| `transformer_kwargs` | jsonb |  | ✓ |  | | JSON object passed to the transformer as kwargs, along with the harvested raw data |
+| `disabled` | bool |  |  |  | False | True if this ingest config should not be run automatically |
+
+### Source
+A Source is a place metadata comes from.
+
+| Column | Type | Indexed | Nullable | FK | Default | Description |
+|:-------|:----:|:-------:|:--------:|:--:|:-------:|:------------|
+| `name` | text | unique |  |  | | Short name |
+| `long_title` | text | unique |  |  | | Full, human-friendly name |
+| `home_page` | text |  | ✓ |  | | URL |
+| `icon` | image |  | ✓ |  | | Recognizable icon for the source |
+| `user_id` | int |  |  | ✓ | | User with permission to submit data as this source (TODO: replace with django permissions stuff) |
+
+### Harvester
+Each row corresponds to a Harvester implementation in python. (TODO: describe those somewhere)
+
+| Column | Type | Indexed | Nullable | FK | Default | Description |
+|:-------|:----:|:-------:|:--------:|:--:|:-------:|:------------|
+| `key` | text | unique |  |  | | Key that can be used to get the corresponding Harvester subclass |
+| `date_created` | datetime |  |  |  | now (on insert) | |
+
+### Transformer
+Each row corresponds to a Transformer implementation in python. (TODO: describe those somewhere)
+
+| Column | Type | Indexed | Nullable | FK | Default | Description |
+|:-------|:----:|:-------:|:--------:|:--:|:-------:|:------------|
+| `key` | text | unique |  |  | | Key that can be used to get the corresponding Transformer subclass |
+| `date_created` | datetime |  |  |  | now (on insert) | |
+
+## Logs
+
+### HarvestLog
+Log entries to track the status of a specific harvester run.
+
+| Column | Type | Indexed | Nullable | FK | Default | Description |
+|:-------|:----:|:-------:|:--------:|:--:|:-------:|:------------|
+| `ingest_config_id` | int |  |  | ✓ | | IngestConfig for this harvester run |
+| `harvester_version` | text |  |  |  | | Current version of the harvester in format 'x.x.x' |
```

Comment:
Version might need to have a different format too it. I'd like it to be sortable. It fails to order `0.0.10` and `0.0.9` correctly. We could just pad the version up to 3 digits or something and not worry about it?